### DISCUSSION TITLE
Is exclusive-consumer supported in quorum queues?

### DISCUSSION THREAD

**Discussion by rahultaneja1534** [rabbitmq/rabbitmq-server#5546] [Questions]
I asked this question on [Stackoverflow](https://stackoverflow.com/questions/73381937/how-to-properly-register-rabbitmq-exclusive-consumer) and while exploring I figured that when I move the queue type from `quorum` to `classic`, the consumer is registered exclusively which I wanted, but would like to clarify if it's also supported in `quorum` queues. If yes, what else I need to do apart from `Consume` call with `exclusive=true` to make it work? If not, is there a plan to support it?
`Single Active Consumer` may not work in our use-case.

**Comment by lukebakken** [comment]
Have you tried it? https://www.rabbitmq.com/consumers.html#exclusivity

**Reply by rahultaneja1534** [reply]
Yes and that's only working when I have `classic` queues, however I would like to use `quorum` queues where it registers all the consumers as `exclusive` and delivers messages in round-robin fashion which is behavior of `non-exclusive` consumers. Please check [https://stackoverflow.com/questions/73381937/how-to-properly-register-rabbitmq-exclusive-consumer](https://stackoverflow.com/questions/73381937/how-to-properly-register-rabbitmq-exclusive-consumer) where I explained the situation with example in `golang`

**Reply by lukebakken** [reply]
Thanks for pointing out your question on stack overflow. I added your code to a repo here:

https://github.com/lukebakken/how-to-properly-register-rabbitmq-exclusive-consumer-73381937

It appears that exclusive consumers are not supported with quorum queues:

![image](https://user-images.githubusercontent.com/514926/185236064-747caf06-7dbc-4418-93e5-2057886c01d1.png)

I'll ask the team about that.

**Reply by rahultaneja1534** [reply]
Thanks for checking @lukebakken. Just found the docs too - https://www.rabbitmq.com/quorum-queues.html#feature-matrix
So it's not supported as of now. The only question remains is if there is a plan to support it in the future.

**Reply by lukebakken** [reply]
No plans at this time. @michaelklishin responded with:

> It's just not used widely enough and semantically conflicts with Single Active Consumer. I would say people should use Single Active Consumer.

**Reply by michaelklishin** [reply]
Yup, you will mostly likely find that Single Active Consumer would do what you want with the extra redundancy/resiliency.
